cmd/access/node_builder/access_node_builder.go

```diff
@@ -1818,6 +1826,25 @@ func (builder *FlowAccessNodeBuilder) Build() (cmd.Node, error) {
 
 			return versionControl, nil
 		}).
+		Component("stop control", func(node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+			if !builder.stopControlEnabled {
+				return &module.NoopReadyDoneAware{}, nil
+			}
+
+			stopControl := stop.NewStopControl(
+				builder.Logger,
+			)
+
+			builder.versionControl.AddVersionUpdatesConsumer(stopControl.OnVersionUpdate)
+
+			if builder.executionDataIndexingEnabled {
+				stopControl.RegisterHeightRecorder(builder.ExecutionIndexer)
```

Comment:
the indexer component is launched as a `DependableComponent` which behaves differently from regular `Component`. Instead of blocking until it finishes, it is started async, and waiting for dependencies is done explicitly. This was needed because loading the database can take over an hour during bootstrapping, and we didn't want to block the rest of the system for that whole time.

As a result, `builder.ExecutionIndexer` is typically not available while starting other components unless they also use `DependableComponent` and explicitly add it as one of their dependencies.